lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -177,8 +178,13 @@ def add_risk_data(post, options)
         end
       end
 
+      def add_stored_credentials(post, payment, options)
+        add_shopper_interaction(post, payment, options)
+        add_recurring_processing_model(post, options)
+      end
+
       def add_shopper_interaction(post, payment, options={})
-        if (payment.respond_to?(:verification_value) && payment.verification_value) || payment.is_a?(NetworkTokenizationCreditCard)
+        if options.dig(:stored_credential, :initial_transaction) || (payment.respond_to?(:verification_value) && payment.verification_value) || payment.is_a?(NetworkTokenizationCreditCard)
```

Comment:
Could `.dig` replace the second condition?

Reply:
I wish! `payment` is not a hash, so it cannot.

Reply:
orite